compiler/rustc_session/src/config.rs

```diff
@@ -2532,6 +2532,19 @@ pub fn build_session_options(early_dcx: &mut EarlyDiagCtxt, matches: &getopts::M
     let debuginfo = select_debuginfo(matches, &cg);
     let debuginfo_compression = unstable_opts.debuginfo_compression;
 
+    if !nightly_options::is_unstable_enabled(matches) {
+        // Check that -Clinker-features=[-+]lld is not used anywhere else than on x64
+        // without -Zunstable-options.
+        if !cg.linker_features.disabled.is_empty() || !cg.linker_features.enabled.is_empty() {
```

Comment:
Let's actually check for lld instead of `!is_empty()`, I was thinking of something like `.contains(LinkerFeatures::LLD)` but then it makes it harder to check for `LinkerFeatures::CC` (to ensure it still needs `-Zunstable-options`, otherwise I think this is stabilizing +/-cc) with the 3 levels of nested negations -- so I suggest trying to instead copy the stable/unstable setup for LinkSelfContained whose check is 50 lines above, with `are_unstable_variants_set` (taking the target as parameter in our case). I think it should be easier that way. 

We can also adjust the error message below to that effect.

Reply:
I wanted to have a function like `are_unstable_variants_set`, I just thought that it's a bit weird that some remote function would be doing the check, and then the error message is generated here in `config.rs`, seems easy to be desynced (but not a big concern).

Reply:
The remote function wouldn't be doing the check, only returning whether something is stable or not; and the error in config is just checking that something unstable isn't called on stable? Do you have an example of a desync?

Reply:
Marking as resolved as we've discussing this entire PR on zulip for a while 👍